drivers/google/compute_util.go

```diff
@@ -292,6 +297,10 @@ func (c *ComputeUtil) createInstance(d *Driver) error {
 		},
 	}
 
+	if d.ThreadsPerCore == 1 || d.ThreadsPerCore == 2 {
```

Comment:
The idea is to pass only valid arguments to this parameter, otherwise gcloud throws an error and creation of instance is failing. Any error thrown during creation process leads to a dangling instance in `docker-machine ls`. So here we ensure that only valid parameter is passed, otherwise `gcloud` runs without `"google-threads-per-core` at all. If this parameter is omitted by default `gcloud compute instances` create instance with value 2.